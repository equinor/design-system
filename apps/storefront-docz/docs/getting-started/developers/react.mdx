---
title: Developers
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
