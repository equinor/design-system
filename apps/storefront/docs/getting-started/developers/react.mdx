---
title: Developers
metaTitle: Developers react
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
