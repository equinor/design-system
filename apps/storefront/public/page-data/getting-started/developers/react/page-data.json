{"componentChunkName":"component---docs-getting-started-developers-react-mdx","path":"/getting-started/developers/react/","result":{"pageContext":{"frontmatter":{"title":"Developers","metaTitle":"Developers react","tabs":["Introduction","CSS framework","React","Figma"],"mode":"draft","route":"/getting-started/developers/react/"},"entry":{"id":"6f9784da00517d57e4627269567073e6","filepath":"docs/getting-started/developers/react.mdx","fullpath":"/Users/vnys/Kode/storefront-replace/apps/storefront/docs/getting-started/developers/react.mdx","route":"/getting-started/developers/react/","slug":"docs-getting-started-developers-react","name":"React","menu":"","headings":[]}}}}