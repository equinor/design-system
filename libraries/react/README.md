@eds-ui/react
===============

1. `npm i -g pnpm`
1. `pnpm install`
1. `pnpm recursive install`

## Run storybook

`pnpm start`

## Build modules

`pnpm recursive run build`
