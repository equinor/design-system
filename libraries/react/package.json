{
  "name": "@eds-ui/react",
  "private": true,
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "storybook": "start-storybook"
  },
  "keywords": [],
  "author": "",
  "license": "AGPL",
  "dependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "@babel/preset-react": "^7.0.0",
    "@storybook/react": "^5.0.11",
    "babel-plugin-styled-components": "^1.10.0",
    "enzyme": "^3.9.0",
    "enzyme-adapter-react-16": "^1.12.1",
    "jest": "^24.8.0",
    "jest-styled-components": "^6.3.1",
    "rollup": "^1.11.3",
    "rollup-plugin-babel": "^4.3.2",
    "rollup-plugin-commonjs": "^9.3.4",
    "rollup-plugin-node-resolve": "^4.2.3",
    "styled-components": "^4.2.0"
  },
  "peerDependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "styled-components": "^4.2.0",
    "prop-types": "^15.7.2"
  },
  "jest": {
    "verbose": true,
    "setupFilesAfterEnv": [
      "<rootDir>/testSetup.js"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "transformIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "haste": {
      "providesModuleNodeModules": [
        ".*"
      ]
    },
    "roots": [
      "packages/node_modules"
    ]
  }
}
