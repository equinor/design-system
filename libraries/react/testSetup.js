import React from 'react'
import Enzyme from 'enzyme'
import 'jest-styled-components'
import styled from 'styled-components'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })
