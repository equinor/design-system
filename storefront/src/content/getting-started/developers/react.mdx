---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
mode: draft
---


:warning: Information coming soon
