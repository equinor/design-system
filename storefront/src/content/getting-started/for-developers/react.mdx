---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
---

Developers React
