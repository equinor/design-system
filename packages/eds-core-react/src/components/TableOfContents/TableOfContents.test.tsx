/* eslint-disable no-undef */
import { render, cleanup, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import { axe } from 'jest-axe'
import styled from 'styled-components'
import { TableOfContents } from '.'

const { LinkItem } = TableOfContents

const StyledTableOfContents = styled(TableOfContents)`
  background: red;
`

const ScrollContainer = styled.div`
  height: 2000px;
  width: 500px;
`

afterEach(cleanup)

describe('TableOfContents', () => {
  it('Matches snapshot', () => {
    const { asFragment } = render(
      <TableOfContents>
        <LinkItem>Anchor</LinkItem>
      </TableOfContents>,
    )

    expect(asFragment()).toMatchSnapshot()
  })
  it('Should pass a11y test', async () => {
    const { container } = render(<TableOfContents />)
    expect(await axe(container)).toHaveNoViolations()
  })
  it('Should pass a11y test with link', async () => {
    const { container } = render(
      <TableOfContents>
        <LinkItem>Anchor</LinkItem>
      </TableOfContents>,
    )
    expect(await axe(container)).toHaveNoViolations()
  })
  it('Can extend the css for the component', () => {
    render(<StyledTableOfContents>Stuff</StyledTableOfContents>)
    const toc = screen.getByRole('navigation')
    expect(toc).toHaveStyleRule('background', 'red')
  })

  it('Can have fixed position in container', () => {
    const testId = 'toc-test'
    render(
      <ScrollContainer>
        <TableOfContents data-testid={testId} sticky>
          Content
        </TableOfContents>
      </ScrollContainer>,
    )

    expect(screen.queryByTestId(testId)).toHaveStyleRule('position', 'fixed')
    expect(screen.queryByTestId(testId)).toHaveStyleRule('top', '32px')
    expect(screen.queryByTestId(testId)).toHaveStyleRule('right', '32px')
  })

  it('Has provided label', () => {
    const label = 'Some toc label'
    render(
      <TableOfContents label={label}>
        <LinkItem>Anchor</LinkItem>
      </TableOfContents>,
    )
    expect(screen.getByText(label)).toBeDefined()
  })

  it('Has all provided content', () => {
    const testIdItem = 'toc-test-linkitem'

    render(
      <TableOfContents>
        <LinkItem data-testid={testIdItem}>
          <p>TestItem</p>
        </LinkItem>
      </TableOfContents>,
    )
    expect(screen.getByTestId(testIdItem)).toBeDefined()
  })
})
