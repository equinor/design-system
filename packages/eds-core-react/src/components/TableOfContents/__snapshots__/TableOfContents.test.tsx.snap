// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableOfContents Matches snapshot 1`] = `
<DocumentFragment>
  .c3 {
  margin: 0;
  color: var(--eds_paragraph__body_short_color,rgba(61,61,61,1));
  font-family: Equinor;
  font-size: 1.000rem;
  font-weight: 400;
  line-height: 1.250em;
  text-align: left;
}

.c1 {
  margin: 0;
  color: var(--eds_paragraph__overline_color,rgba(61,61,61,1));
  font-family: Equinor;
  font-size: 0.625rem;
  font-weight: 500;
  line-height: 1.600em;
  -webkit-letter-spacing: 0.069em;
  -moz-letter-spacing: 0.069em;
  -ms-letter-spacing: 0.069em;
  letter-spacing: 0.069em;
  text-transform: uppercase;
  text-align: left;
}

.c0 {
  margin-top: 48px;
  margin-bottom: 32px;
}

.c4 {
  margin: 0;
  padding: 0;
}

.c2 {
  margin: 0;
  color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
  font-family: Equinor;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.143em;
  text-align: left;
}

.c5 {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c5 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  color: var(--eds_interactive_primary__resting,rgba(0,112,121,1));
  font-family: Equinor;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.143em;
  text-align: left;
  padding-left: 18px;
  padding-top: 10px;
  padding-right: 18px;
  padding-bottom: 10px;
  height: 1.143em;
  width: calc(189px - 36px);
  display: block;
  position: relative;
}

.c5 a svg {
  fill: var(--eds_interactive_primary__resting,rgba(0,112,121,1));
  margin-right: 8px;
  vertical-align: text-bottom;
}

.c5 a span {
  max-width: 115px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  vertical-align: text-bottom;
  display: inline-block;
}

.c5 a:focus {
  outline: 1px dashed var(--eds_interactive__focus,rgba(0,112,121,1));
  outline-offset: 2px;
}

.c5 a:active {
  outline: none;
}

@media (hover:hover) and (pointer:fine) {
  .c5 a:hover {
    margin: 0;
    color: var(--eds_interactive_primary__hover,rgba(0,79,85,1));
    font-family: Equinor;
    font-size: 0.875rem;
    font-weight: 500;
    line-height: 1.143em;
    text-align: left;
    background: var(--eds_interactive_primary__hover_alt,rgba(222,237,238,1));
  }

  .c5 a:hover svg {
    fill: var(--eds_interactive_primary__hover,rgba(0,79,85,1));
  }
}

<nav
    class="c0"
    label=""
  >
    <p
      class="c1 c2"
    />
    <ul
      class="c3 c4"
    >
      <li
        class="c5"
      >
        Anchor
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
